ROX-20769: Add changelog entry about major version upgrade (#93434)
openshift/openshift-docs: release_notes/48-release-notes.adoc

This release adds improvements related to the following components and concepts:

//ROX-20769
[id="central-db-postgresql_15_{context}"]
=== Central DB uses PostgreSQL 15

Central DB component is using PostgreSQL 15 now, and {product-title-short} 4.8
supports this version for external databases as well. A new installation with
an internal database will use this version by default now. When upgrading an
existing cluster to {product-title-short} 4.8, Central DB will do an in-place
upgrade of the data collected so far.

> [NOTE]

Reviewer: I would not have these bullets as a note (multiple lines in notes does not look the best) and have it be a sentence with bullets instead. Maybe move the part about not upgrading the database to here, right before you say "To ensure..." and make that an "important" note:

```
[IMPORTANT]
====
text
====
```

Author: Not quite follow what do you mean by "move the part about not upgrading the database to here"? Which part are you referring to?

====

> To make the database upgrade process smooth, we recommend:

Author: This one is tricky, because "do this to make the upgrade successful" is a too strong statement. The point is that the migration will be successful for almost everyone, we've put lots of efforts into this. Those recommendations are more about avoiding corner cases, where things are getting complicated.

That being said, how about this?
```
When preparing the upgrade to RHACS 4.8, follow these suggestions: 
```

> * Take a backup before upgrading to {product-title-short} 4.8

Reviewer: ```suggestion
* Back up the database before upgrading to {product-title-short} 4.8
```
(Or if you want them to make a backup of everything, "Back up the system before upgrading....")

> * If not using the Operator: check disk space available for the database

Reviewer: ```suggestion
* If you are not upgrading by using the Operator, check the disk space available for the database by viewing the
```

> `rox_central_postgres_available_size_bytes`. For the purposes of the upgrade,

Reviewer: ```suggestion
`rox_central_postgres_available_size_bytes` variable. For the purposes of the upgrade,
```
Not sure if "variable" or "parameter" is correct here, but I think "variable"?

Author: Probably "metric" would be better.

> it should be double of already consumed disk space

Reviewer: ```suggestion
the value should be double the amount of the already-consumed disk space, as shown in the 
```

> `rox_central_postgres_total_size_bytes`. If that is not the case, extend the

Reviewer: ```suggestion
`rox_central_postgres_total_size_bytes` variable. If the value is not correct, extend the
```

> database PVC.

Reviewer: I assume user will know how to do this (?)

Author: I haven't found any detailed instructions for that in ACS documentation, but yes, it's a regular K8s/OpenShift thing they should be aware of.

> * Do not interrupt the upgrade procedure. If that happens, a manual

Reviewer: ```suggestion
* Do not interrupt the upgrade procedure. If the upgrade is interrupted, a manual
```

> * Do not interrupt the upgrade procedure. If that happens, a manual

Reviewer: replied below, you can ignore this :)

intervention could be required to proceed.

> Depending on the amount of data the upgrade may take extra time to finish.

Reviewer: ```suggestion
Depending on the amount of data, the upgrade can take extra time to finish.
```
* Move this sentence to go with previous paragraph that I am recommending to change into an important note.
* If you are curious, you can see the guidance on "may" in [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#may) (use redhat ID to log in)

This release adds improvements related to the following components and concepts:

//ROX-20769
[id="central-db-postgresql_15_{context}"]
=== Central DB uses PostgreSQL 15

The Central DB component now uses PostgreSQL 15, and {product-title-short} 4.8
supports this version for external databases. A new installation with an
internal database now uses this version by default. When upgrading an existing
cluster to {product-title-short} 4.8, Central DB will perform an upgrade of the
data it has collected.

[IMPORTANT]
====
When preparing the upgrade to {product-title-short} 4.8, follow these
suggestions:

> * Back up the database before upgrading to {product-title-short} 4.8

Reviewer: ```suggestion
* Back up the database before upgrading to {product-title-short} 4.8.
```
